Tests/AsyncHTTPClientTests/HTTP2ConnectionTests.swift

```diff
@@ -41,6 +41,28 @@ class HTTP2ConnectionTests: XCTestCase {
             logger: logger
         ).wait())
     }
+    
+    func testConnectionToleratesShutdownEventsAfterAlreadyClosed() {
+        let embedded = EmbeddedChannel()
+        XCTAssertNoThrow(try embedded.connect(to: SocketAddress(ipAddress: "127.0.0.1", port: 3000)).wait())
+        
+        let logger = Logger(label: "test.http2.connection")
+        let connection = HTTP2Connection(
+            channel: embedded,
+            connectionID: 0,
+            decompression: .disabled,
+            delegate: TestHTTP2ConnectionDelegate(),
+            logger: logger
+        )
+        _ = connection.start()
```

Comment:
The problem is that the only way to for the the future returned by the `HTTP2Connection.start()` to fulfill is by sending `ByteBuffer`s of `HTTP2Frame`s through the `Channel`. `HTTP2Frame` de/encoding is currently not publicly available.

Reply:
But we don't need that future, or am I missing something here?

Reply:
We do need it as only its success value contains the `HTTP2Connection` instance. We otherwise don't have access to it and therefore can't call `.shutdown()`.

Reply:
stupid me... makes sense. can we rename it to start0 though. to make sure it is called on el, if someone else internally tries to call it?

Reply:
Sure changed in https://github.com/swift-server/async-http-client/pull/646/commits/ba76fe445fcd25559a564364229c742db8c63316